esphome/components/psram/__init__.py

```diff
@@ -4,15 +4,35 @@
 from esphome.core import CORE
 from esphome.const import (
     CONF_ID,
+    CONF_MODE,
+    CONF_SPEED,
 )
 
 CODEOWNERS = ["@esphome/core"]
 
 psram_ns = cg.esphome_ns.namespace("psram")
 PsramComponent = psram_ns.class_("PsramComponent", cg.Component)
 
+SPIRAM_MODES = {
+    "quad": "CONFIG_SPIRAM_MODE_QUAD",
+    "oct": "CONFIG_SPIRAM_MODE_OCT",
```

Comment:
Maybe full name for it here?
```suggestion
    "octal": "CONFIG_SPIRAM_MODE_OCT",
```

Reply:
SGTM!